exercises/alice_russell/ex02_oo_basics/src/ShapeSorter.cpp

```diff
@@ -0,0 +1,51 @@
+#include "ShapeSorter.h"
+
+bool ShapeSorter::compare_area(Shape* &s1, Shape* &s2)
```

Comment:
A more standard signature for this type of method would be `bool ShapeSorter::compare_area(const Shape &s1, const Shape &s2)`. It reads as two references to a const `Shape`" object so `compare_area` can't change either of them. You would generally not see a signature such as `Shape * &` as copying pointers is basically free and a pointer allows you to change the object it points to.

Reply:
Removing the pointer leads to an error using sort in print_area_descending. Is there a way to make it so the object can't be change but keeping the pointer? Would 
```
bool ShapeSorter::compare_area(const Shape * const s1, const Shape * const s2)
``` 
be sufficient?

Reply:
Ah yes of course - I'd missed the `std::sort` call. Your suggested solution would also work.